apps/gallery/test/marionette/lib/gallery.js

```diff
@@ -310,6 +386,45 @@ Gallery.prototype = {
     }.bind(this));
   },
 
+  switchToGalleryFrame: function() {
+    var galleryFrame = this.client.findElement('iframe[src*="' +
+                                          Gallery.ORIGIN + '"]');
+    this.waitFor(galleryFrame);
+    this.client.switchToFrame(galleryFrame);
```

Comment:
How is switchToFrame() different than switchToApp()?

Am I right in thinking that this switchToGalleryFrame() function can only be called if we have previously done a this.client.switchToFrame() to switch to the system app?  If so, please note that in a comment, or move the whole function into a separate library that represents the system app and action menu.